syntax = "proto3";

package user;

service UserService {
  // Unary request and response
  rpc get (UserRequest) returns (UserResponse);

  // Single request -> Stream response
  // rpc getServerStreamMethod (UserRequest) returns (stream UserResponse);

  // Stream request -> Single response
  // rpc getClientStreamMethod (stream UserRequest) returns (UserResponse);

  // Dualistic requests and responses
  // rpc getBidirectionalMethod (stream UserRequest) returns (stream UserResponse);
}

message User {
  uint32 id = 1;
  string name = 2;
  UserStatus status = 3;
  repeated string hobby_list = 4;
}

enum UserStatus {
  NONE = 0;
  ACTIVE = 1;
  INACTIVE = 2;
}

message UserRequest {
  uint32 id = 1;
}

message UserResponse {
  bool error = 1;
  string message = 2;
  User user = 3;
}

